### PR TITLE
Add Dataset integration tests - Dataset CRUD + actions outside of data.all

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Feature

### Detail
It implements some tests for s3_datasets (check full list in #1358)
### For fresh deployments
- [x] Create Dataset
- [x] Import Dataset --> IMPORTANT: See below details on AWS actions for testing
- [x] List Datasets
- [X] Get Dataset
- [x] Edit Dataset
- [x] Delete dataset - decision: I only added explicit test for delete_unauthorized since delete_dataset is covered in the fixtures and it takes a long time to deploy + delete. If needed we can introduce the test for better reporting
- [X] Access dataset assume role url
- [X] Generate dataset access token
- [X] Dataset upload data presigned url
- [X] Backwards compatibility - update dataset
- [X] Backwards compatibility - import dataset

🔦 **AWS actions outside of data.all**
There are some actions that in real life are performed outside of data.all.  To run the tests we need to either perform this actions manually before the tests are executed or we can use AWS SDK to automate them. Most important actions performed outside of data.all.
- Creation of consumption roles
- Creation of imported dataset bucket, kms key and glue database *IN THIS PR
- Create VPCs for Notebooks
- Validate shares - we assume the share request role for this

To create resources we need to assume a role in the environment account. We could assume the pivot role, but then we need to ensure that it has CreateBucket... permissions; which is not the case. I have opted to create a separate isolated role `dataall-integration-tests-role` as part of the environment stack ONLY when we are creating environments during integration testing. As part of the global config of environments users can use the boto3 session of this role to perform direct AWS calls in the environment account.

In https://github.com/data-dot-all/dataall/pull/1382/files we discussed some alternatives. In this PR we use the `environmentType` variable in the environment model, which was not used for anything (it always defaulted to Data environments). 
API call create environment (input: environmentType = IntegrationTesting) ---> in environment stack we check the type of environment and deploy the integration test role.

Then we use an SSM parameter to read the tooling account id needed for the assume role trust policy

### Relates
- https://github.com/data-dot-all/dataall/issues/1358

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
##Testing locally
- Create Environment outside of testing - does not create testing integration role
- testing:

File: tests_new/integration_tests/modules/s3_datasets/test_s3_dataset.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    dataset_uri = persistent_imported_kms_s3_dataset1.datasetUri
+    target_type = 'dataset'
+    check_stack_ready(client1, env_uri=env_uri, stack_uri=stack_uri, target_uri=dataset_uri, target_type=target_type)
+    update_stack(client1, dataset_uri, 'dataset')
+    # wait for stack to move to "in_progress" state
+
+    # TODO: Come up with better way to handle wait in progress if applicable
+    # Use time.sleep() instead of poller b/c of case where no changes founds  (i.e. no update required)
+    # check_stack_in_progress(client1, env_uri, stack_uri)
+    time.sleep(10)
```
Comment (dlpzx) [inline tests_new/integration_tests/modules/s3_datasets/test_s3_dataset.py:153]: Idea: Check the events that come from update_stack

File: tests_new/integration_tests/modules/s3_datasets/test_s3_dataset.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    dataset_uri = persistent_imported_sse_s3_dataset1.datasetUri
+    target_type = 'dataset'
+    check_stack_ready(client1, env_uri=env_uri, stack_uri=stack_uri, target_uri=dataset_uri, target_type=target_type)
+    update_stack(client1, dataset_uri, target_type)
+    # wait for stack to move to "in_progress" state
+
+    # TODO: Come up with better way to handle wait in progress if applicable
+    # Use time.sleep() instead of poller b/c of case where no changes founds  (i.e. no update required)
+    # check_stack_in_progress(client1, env_uri, stack_uri)
+    time.sleep(10)
```
[the remark this answers is not in the document]
Comment (petrkalos) [reply tests_new/integration_tests/modules/s3_datasets/test_s3_dataset.py:132]: You can disregard the above comments, when the are `no changes` in the stack then `cdk deploy` doesn't generate a new events. As an alternative we can use the `getStackLogs`

[the remark this answers is not in the document]
Comment (dlpzx) [reply tests_new/integration_tests/modules/s3_datasets/test_s3_dataset.py]: For the time being I increased the sleep time to give ECS/CFN time to update

**Review by petrkalos — APPROVED** [review]
I'd only suggest to increase the timeout to 2 minutes as privisioning+running of cdkproxy takes about 1 minute

**Pull request merged**